src/CollectionIndex.js

```diff
@@ -9,51 +10,13 @@ class CollectionIndex {
         return this._index[key]
     }
     async find(query) {
-        var results = [];
-        var index = this._index;
-        for (var id in index) {
-            var doc = index[id];
-            var match = true;
-
-            for (var key in query) {
-                if (!doc[key]) {
-                    match = false;
-                    break;
-                }
-                if (doc[key] !== query[key]) {
-                    match = false;
-                    break;
-                }
-            }
-
-            if (match) {
-                results.push(doc);
-            }
-        }
-        return results;
+        let res = parseAndFind(query, this._index, false)
+        return res
```

Comment:
Possible issue here. How does this affect when we want to use an on disk index or other optimized index?

Reply:
I'm not aware of that. How should this be written?

Reply:
I am not sure at the moment. We don't have any formal API for index/store. I think for now its alright.